src/recordhost.cpp

```diff
@@ -0,0 +1,349 @@
+/*
+    SPDX-FileCopyrightText: Lieven Hey <lieven.hey@kdab.com>
+    SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+
+    SPDX-License-Identifier: GPL-2.0-or-later
+*/
+
+#include <QDebug>
+#include <QDir>
+#include <QFileInfo>
+#include <QProcess>
+#include <QStandardPaths>
+#include <QThread>
+
+#include <KShell>
+#include <KUser>
+#include <ThreadWeaver/ThreadWeaver>
+
+#include <fstream>
+#include <sys/stat.h>
+
+#include "recordhost.h"
+#include "settings.h"
+#include "util.h"
+
+#include "hotspot-config.h"
+
+namespace {
+QByteArray perfOutput(const QString& perfPath, const QStringList& arguments)
+{
+    // TODO handle error if man is not installed
+    QProcess process;
+
+    auto reportError = [&]() {
+        qWarning() << "Failed to run perf" << process.arguments() << process.error() << process.errorString()
+                   << process.readAllStandardError();
+    };
+
+    QProcessEnvironment env = QProcessEnvironment::systemEnvironment();
+    env.insert(QStringLiteral("LANG"), QStringLiteral("C"));
+    process.setProcessEnvironment(env);
+
+    QObject::connect(&process, &QProcess::errorOccurred, &process, reportError);
+    process.start(perfPath, arguments);
+    if (!process.waitForFinished(1000) || process.exitCode() != 0)
+        reportError();
+    return process.readAllStandardOutput();
+}
+
+QByteArray perfRecordHelp(const QString& perfPath)
+{
+    QByteArray recordHelp = [&perfPath]() {
```

Comment:
simplify, the outer lambda/IILE isn't needed anymore, now that we don't store the `recordHelp` in a static anymore